universal-login-sdk/test/integration/core/AggregateBalanceObserver.ts

```diff
@@ -0,0 +1,87 @@
+import chai, {expect} from 'chai';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+import {providers, Wallet, Contract, utils} from 'ethers';
+import {createMockProvider, getWallets, deployContract} from 'ethereum-waffle';
+import {BalanceChecker, TokenDetails, ETHER_NATIVE_TOKEN, TEST_ACCOUNT_ADDRESS, waitUntil} from '@universal-login/commons';
+import {AggregateBalanceObserver} from '../../../lib/core/observers/AggregateBalanceObserver';
+import {BalanceObserver} from '../../../lib/core/observers/BalanceObserver';
+import MockToken from '@universal-login/contracts/build/MockToken.json';
+
+chai.use(sinonChai);
+
+describe('INT: AggregateBalanceObserver', () => {
+  let provider: providers.Provider;
+  let balanceChecker: BalanceChecker;
+  let balanceObserver: BalanceObserver;
+  let aggregateBalanceObserver: AggregateBalanceObserver;
+  let wallet: Wallet;
+  let mockToken: Contract;
+  let observedTokens: TokenDetails[] = [];
+
+  const priceOracle = {
+    getTokenPrice: (tokenSymbol: string) => {
+      return Promise.resolve(1000);
```

Comment:
I like this prediction.

Reply:
I not sure if it works, can you add a `setTimeout(...,1)` to make sure it works?